pages/academy.php

```diff
@@ -0,0 +1,499 @@
+            }
+
+			// If sensei
+            if (!isset($lesson_sensei)) {
+                throw new RuntimeException('Target player is not a valid sensei!');
+            }
+
+			// If active
+			if (!SenseiManager::isActiveSensei($lesson_data['sensei_id'], $system)) {
+				throw new RuntimeException('Target player is not a valid sensei!');
+            }
+            $lesson_sensei += SenseiManager::getSenseiUserData($lesson_data['sensei_id'], $system);
+
+			// If in village
+			if ($player->village->name != $lesson_sensei['village']) {
+				throw new RuntimeException('Target player is not a valid sensei!');
+            }
+
+			// If accepting lessons
+            if (!((bool) $lesson_sensei['enable_lessons'] && (bool) $lesson_sensei['accept_students'])) {
+                throw new RuntimeException('Target player is not a valid sensei!');
+            }
+
+			// If has slot available
+            if (!SenseiManager::hasSlot($lesson_data['sensei_id'], $system)) {
+                throw new RuntimeException('Target player is not a valid sensei!');
+            }
+			$lesson_cost = SenseiManager::getLessonCostForPlayer($player, $system)[$lesson_data['train_type']];
+			$lesson_duration = SenseiManager::getLessonDurationForPlayer($player, $system)[$lesson_data['train_type']];
+            switch ($_POST['lesson_stat']) {
+                case 'ninjutsu_skill':
+                    $lesson_modifier = SenseiManager::getLessonModifier($player->ninjutsu_skill, $player->exp, $lesson_sensei['ninjutsu_skill'], $lesson_sensei['exp'], $lesson_sensei['specialization'] == 'ninjutsu_skill' ? true : false);
+                    break;
+                case 'taijutsu_skill':
+                    $lesson_modifier = SenseiManager::getLessonModifier($player->taijutsu_skill, $player->exp, $lesson_sensei['taijutsu_skill'], $lesson_sensei['exp'], $lesson_sensei['specialization'] == 'taijutsu_skill' ? true : false);
+                    break;
+                case 'genjutsu_skill':
+                    $lesson_modifier = SenseiManager::getLessonModifier($player->genjutsu_skill, $player->exp, $lesson_sensei['genjutsu_skill'], $lesson_sensei['exp'], $lesson_sensei['specialization'] == 'genjutsu_skill' ? true : false);
+                    break;
+                case 'bloodline_skill':
+					if ($player->bloodline_id == $lesson_sensei['bloodline_id']) {
+						$lesson_modifier = round(SenseiManager::getLessonModifier($player->bloodline_skill, $player->exp, $lesson_sensei['bloodline_skill'], $lesson_sensei['exp'], true) * 100 - 100, 2);
+					} else {
+						throw new RuntimeException('Invalid stat!');
+                    }
+                    break;
+                case 'speed':
+                    $lesson_modifier = SenseiManager::getLessonModifier($player->speed, $player->exp, $lesson_sensei['speed'], $lesson_sensei['exp'], $lesson_sensei['specialization'] == 'speed' ? true : false);
```

Comment:
Totally optional, but we could be slightly more specific by calling this `$lesson_gain_multiplier`  or similar

Reply:
Sounds good to me.